### PR TITLE
🎸Features: Search movie

### PR DESCRIPTION
**Pull request by danyouknowme**
Implement search movies
  - Get search movies sorted by popularity
  - Create or update a search movie

### CODE REVIEW

File: handler/search_movie.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"net/http"
+
+	model "github.com/BounkBU/doonungfangpleng/models"
+	"github.com/BounkBU/doonungfangpleng/service"
+	"github.com/gin-gonic/gin"
+)
+
+type SearchMovieHandler struct {
+	searchMovieService *service.SearchMovieService
+}
```
Comment (abounk) [inline handler/search_movie.go]: this should be private

File: repository/search_movie.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package repository
+
+import (
+	model "github.com/BounkBU/doonungfangpleng/models"
+	"github.com/jmoiron/sqlx"
+)
+
+type SearchMovieRepository struct {
+	db *sqlx.DB
+}
```
Comment (abounk) [inline repository/search_movie.go]: this should be private

File: service/search_movie.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	log "github.com/sirupsen/logrus"
+
+	model "github.com/BounkBU/doonungfangpleng/models"
+	"github.com/BounkBU/doonungfangpleng/repository"
+)
+
+type SearchMovieService struct {
+	searchMovieRepository *repository.SearchMovieRepository
+}
```
Comment (abounk) [inline service/search_movie.go]: This should be private

File: repository/search_movie.go (hunk 2 above)
Comment (danyouknowme) [reply repository/search_movie.go]: fixed

File: handler/search_movie.go (hunk 1 above)
Comment (danyouknowme) [reply handler/search_movie.go]: fixed

File: service/search_movie.go (hunk 3 above)
Comment (danyouknowme) [reply service/search_movie.go]: fixed

**Pull request merged**